### ISSUE TITLE
MappedFile can not open a read-only file. Throws exception 100% of the time.

### ISSUE THREAD

**Issue by jonross-st** [OpenHFT/Chronicle-Bytes#133]
simple code 

```
    val f = File("/tmp/map.txt")
    val inFile = MappedFile.readOnly(f)
    val buf = inFile.acquireBytesForRead(ReferenceOwner.INIT, 0 )
```
throws this exception

```
Exception in thread "main" java.lang.IllegalStateException: Read Only
	at net.openhft.chronicle.bytes.ReadOnlyMappedBytesStore.checkReadOnly(ReadOnlyMappedBytesStore.java:176)
	at net.openhft.chronicle.bytes.ReadOnlyMappedBytesStore.bytesForWrite(ReadOnlyMappedBytesStore.java:166)
	at net.openhft.chronicle.bytes.ReadOnlyMappedBytesStore.bytesForWrite(ReadOnlyMappedBytesStore.java:31)
	at net.openhft.chronicle.bytes.BytesStore.bytesForRead(BytesStore.java:128)
	at net.openhft.chronicle.bytes.MappedFile.acquireBytesForRead(MappedFile.java:419)
	at com.sporttrade.exchange.AppKt.main(App.kt:93)
```

It seams that `ReadOnlyMappedByteStore` doesn't implement the acquireBytesForRead(), it lets the parent classes do that, all the way up to `ByteStore` which has this code

```
   default Bytes<Underlying> bytesForRead() throws IllegalStateException {
        try {
            return bytesForWrite()
                    .readLimit(writeLimit())
                    .readPosition(start());
        } catch (BufferUnderflowException e) {
            throw new IllegalStateException(e);
        }
    }
```

Which will fail 100% of the time as it's trying to create a bytesForWrite() on a readl-only store.

So, AFAICT there is no way to create a real-only memory mapped file using MappedFile.

**Comment by dpisklov** [comment]
I assume the file doesn't exist prior to executing this code. If so it's expected behaviour - if you force read only mode, there's no way the file can be created and hence there's no file to map.

**Comment by jonross09** [comment]
I created a pull request that implements a unit test that shows the bug.

**Comment by peter-lawrey** [comment]
Thank you for reporting this. This has been fixed.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.43](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.43), [BOM-2.20.67](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.67)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
